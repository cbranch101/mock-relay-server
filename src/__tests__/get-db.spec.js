import getCollection from '../get-collection'

const userData = [
    {
        id: '1',
        name: 'tom',
    },
    {
        id: '2',
        name: 'larry',
    },
]

const postData = [
    {
        id: '1',
        text: 'the first post',
    },
    {
        id: '2',
        text: 'the second post',
    },
]

test('can find item by ID', async () => {
    const posts = getCollection(postData)
    const foundItem = await posts.findById('2')
    expect(foundItem).toMatchSnapshot()
})

test('can find and insert items', async () => {
    const users = getCollection(userData)
    const foundUsers = await users.find()
    expect(foundUsers).toMatchSnapshot()
    const createdUser = await users.insert({
        name: 'john',
    })
    expect(createdUser).toMatchSnapshot()
    const afterCreateUsers = await users.find()
    expect(afterCreateUsers).toMatchSnapshot()
})

test('can update users', async () => {
    const users = getCollection(userData)
    const updatedUser = await users.update('1', {
        name: 'updatedName',
    })
    expect(updatedUser).toMatchSnapshot()
    const afterUpdateUsers = await users.find()
    expect(afterUpdateUsers).toMatchSnapshot()
})
