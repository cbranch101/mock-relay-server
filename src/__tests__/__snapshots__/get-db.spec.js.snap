// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can find and insert items 1`] = `
Array [
  Object {
    "id": "1",
    "name": "tom",
  },
  Object {
    "id": "2",
    "name": "larry",
  },
]
`;

exports[`can find and insert items 2`] = `
Object {
  "id": "c975ab53-1c23-5eaa-bcd9-0f1dd7efa768",
  "name": "john",
}
`;

exports[`can find and insert items 3`] = `
Array [
  Object {
    "id": "1",
    "name": "tom",
  },
  Object {
    "id": "2",
    "name": "larry",
  },
  Object {
    "id": "c975ab53-1c23-5eaa-bcd9-0f1dd7efa768",
    "name": "john",
  },
]
`;

exports[`can find item by ID 1`] = `
Object {
  "id": "2",
  "text": "the second post",
}
`;

exports[`can update users 1`] = `
Object {
  "id": "1",
  "name": "updatedName",
}
`;

exports[`can update users 2`] = `
Array [
  Object {
    "id": "1",
    "name": "updatedName",
  },
  Object {
    "id": "2",
    "name": "larry",
  },
]
`;
